apps/src/blockly/addons/contextMenu.ts

```diff
@@ -386,21 +460,75 @@ function setAllWorkspacesTheme(
     }
   });
 }
-
+/**
+ * Unregister some default options. We do this either because the options are needlessly
+ * advanced for our target users or because the options have undesired impacts.
+ */
 function unregisterDefaultOptions() {
-  // Remove some default context menu options, if they are present.
   // This needs to be wrapped in a try for now because our GoogleBlocklyWrapperTest.js
   // is not correctly cleaning up its state.
   try {
+    // Option to collapse or expand a block.
     GoogleBlockly.ContextMenuRegistry.registry.unregister(
       'blockCollapseExpand'
     );
+    // Option to open help for a block. Overrided to use our documentation.
     GoogleBlockly.ContextMenuRegistry.registry.unregister('blockHelp');
+    // Option to use inline inputs .
     GoogleBlockly.ContextMenuRegistry.registry.unregister('blockInline');
-    // cleanUp() doesn't currently account for immovable blocks.
+    // Option to clean up blocks. cleanUp() doesn't currently account for immovable blocks.
     GoogleBlockly.ContextMenuRegistry.registry.unregister('cleanWorkspace');
+    // Option to collapse all blocks on a workspace.
     GoogleBlockly.ContextMenuRegistry.registry.unregister('collapseWorkspace');
+    // Option to expand all blocks on a workspace.
     GoogleBlockly.ContextMenuRegistry.registry.unregister('expandWorkspace');
+    // Option to delete all blocks on a workspace.
     GoogleBlockly.ContextMenuRegistry.registry.unregister('workspaceDelete');
   } catch (error) {}
 }
+
+function registerCustomBlockOptions() {
+  // Each option has a "weight": a number that determines the sort order of the option.
+  // Options with higher weights appear later in the context menu.
+
+  // Expected registered block options (from Core and plugins):
+  // 0: blockCopyToStorage
+  // 2: blockComment
+  // 5: blockDisable
+  // 6: blockDelete
+
+  // Our custom options should show below the registered default options.
+  let nextWeight = 7;
+
+  // Custom block options. We increment the weight for each so they are automatically
+  // sorted in the order listed here.
+  registerHelp(nextWeight++);
+  registerDeletable(nextWeight++);
+  registerMovable(nextWeight++);
+  registerEditable(nextWeight++);
+  registerShadow(nextWeight++);
+  registerUnshadow(nextWeight);
+}
+
+function registerCustomWorkspaceOptions() {
```

Comment:
can you add comment(s) explaining the difference between workspace and block options?

Reply:
Yes, but it really comes down to whether it's an option for when you right-click a block vs right-click the workspace.